aten/src/ATen/TensorIterator.cpp

```diff
@@ -40,10 +40,26 @@ namespace {
 
 inline void get_base_ptrs(char** ptrs, ArrayRef<OperandInfo> operands) {
   std::transform(operands.begin(), operands.end(), ptrs, [](const OperandInfo& op) {
-    return static_cast<char*>(op.data);
+    return static_cast<char*>(op.mutable_data());
   });
 }
 
+inline void get_mutable_base_ptrs(char** ptrs, ArrayRef<OperandInfo> operands) {
+  for (const auto arg : c10::irange(operands.size())) {
+    if (!operands[arg].is_const()) {
+      *ptrs++ = static_cast<char*>(operands[arg].mutable_data());
```

Comment:
It's true that the client of TI does have to know how many inputs were added with `add_const_input`, since it changes the size of the `mutable_data` and `const_data` buffers of the loop function.

It sounds like (correct me if I'm wrong) an alternative to this that you suggested is to just put all the pointers into the mutable `char** data`  parameter of the loop function and rely on the user to keep track of which ones are supposed to be const.

So either way, the user would have to keep track of something new.

To me, it seems like it would be easier to keep track of which ones are const by actually putting them in a different buffer. Either way though, the problem I have with giving the client a mutable pointer to data that is supposed to be kept const is that if they accidentally mutate the data, it would be difficult to detect that it's happening